Export path from xml

prince-biswaranjan opened the pull request:
Added functionality to export files to custom paths based on export_paths.xml

prince-biswaranjan committed:
changed the folder structure to use <filename>_<extension> instead of <filename>.<extension>. This ensures that we can add .xlsm to .gitignore files.

prince-biswaranjan committed:
updated README.md to reflect the src/yourfilename_xlsm change

prince-biswaranjan committed:
add sample xml file

prince-biswaranjan committed:
Added functionality to export modules to custom paths based on export_paths.xml file.

cavo789 commented:
Hello pbvirus

Thanks for your code and your interest in my script.

Can you give me some tips to understand why you've decide to rename the output folder from "filename.xlsm" to "filename_xlsm" and the idea behind your XML file ?

Just trying to understand how the script fits your needs.

Thanks and have a nice day.

prince-biswaranjan commented:
Hi Chris,

Filename_xlsm:
I didn't want to check in the .xlsm file as I've some configuration sheets inside that have sensitive data. And adding *.xlsm to gitignore ignores the code files as well. That's why I changed the export folder to _xlsm. My plan is to have an installation script which will create the project file (xlsm) from scratch by creating a new file and add the macros to it.

export_paths.xml:
I've another repository where I put my commonly used macros and make that a submodule for other repositories. That's the reason I added the XML file to specify the path where the vbComponent would be exported. That way I can export the common components to the SubModules directory (and push them to their respective repositories) and the project-specific files to the filename_xlsm folder. Having the export_paths.xml will also help me while importing macros from multiple repositories. This is supposed to work similarly to the package.json file of Node projects.

I wanted to create the PR for my forked repo but raised it for your repo instead. I closed the PR as you don't accept PRs in your repo.

cavo789 commented:
Hello and thanks for your answer

> I wanted to create the PR for my forked repo but raised it for your repo instead. I closed the PR as you don't accept PRs in your repo.

The reason is, just, my script fully answer my need and I'm sharing it with a big pleasure but I don't want to have to manage versions, PR, ... and work to fullfill needs that I don't have ;) The major issue with people doing Open Source isn't the time to write the code (important of course) but the time to manage external things like PRs, coding features for others, ... For that reason, I don't accept PR. So you know why ;)

>Filename_xlsm:
Ok, understood your "workaround".

>export_paths.xml:
Brilliant idea!

I didn't have that need because I've create two addins for Excel so code that can be reused are stored in my addins.

Have a nice day / evening.

Christophe
